lib/reconcile/astarte_dashboard.go

```diff
@@ -163,7 +163,20 @@ func getAstarteDashboardConfigMapData(cr *apiv1alpha1.Astarte, dashboard apiv1al
 	v := getSemanticVersionForAstarteComponent(cr, dashboard.Version)
 	constraint, _ := semver.NewConstraint("< 0.11.0")
 
-	if dashboard.Config.RealmManagementAPIURL == "" {
+	constraint10, _ := semver.NewConstraint(">= 1.0.0")
+	isAstarte10 := constraint10.Check(v)
+
+	if isAstarte10 {
+		// Astarte 1.0+ just needs astarte_api_url, and single API urls only if they're explicit
+		dashboardConfig["astarte_api_url"] = getBaseAstarteAPIURL(cr)
+		// We also enable Flow preview for Astarte 1.0+
+		dashboardConfig["enable_flow_preview"] = true
```

Comment:
this should be enabled only if flow is deployed

Reply:
Flow is always deployed for Astarte 1.0+:
https://github.com/astarte-platform/astarte-kubernetes-operator/blob/master/lib/reconcile/astarte_generic_api.go#L159

Reply:
This isn't true @rbino . That check happens **after** this: https://github.com/astarte-platform/astarte-kubernetes-operator/blob/master/lib/reconcile/astarte_generic_api.go#L142

So you should always check if the component is going to be deployed or not through the usual helper functions.